pkgs/native_toolchain_c/lib/src/cbuilder/linker_options.dart

```diff
@@ -116,6 +116,8 @@ extension LinkerOptionsExt on LinkerOptions {
             ..._symbolsToKeep?.map((symbol) => '-u,_$symbol') ?? [],
             if (stripDebug) '-S',
             if (gcSections) '-dead_strip',
+            if (_generateLinkerScript && _symbolsToKeep != null)
+              '-exported_symbols_list,${_createMacSymbolList(_symbolsToKeep)}',
```

Comment:
Should we pass the value of `linkerScript` to `-exported_symbols_list`, if specified? We do this for the other platforms.